ToDictionary on Mono causes "The type 'Dictionary<,>' is defined in an assembly that is not referenced"

Verified sample repro:
```cs
var d = Array.Empty<int>().ToDictionary(i => i);
```

If I use any overload of `System.Linq.Enumerable.ToDictionary`, Cake 0.22–0.24 (on Travis using Mono 5.2.0 on both Ubuntu Trusty and OS X 10.12) gives this error:

> error CS0012: The type 'Dictionary<,>' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Collections, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

Code that initially triggered it: https://github.com/nunit/nunit/commit/5d3fe04d65889729e9529a93624992d0f0e16ddd#diff-a81006b28a9970a7e6644efe520431a5R9

Failing build log:
https://travis-ci.org/nunit/nunit/jobs/322775654#L1909 (Ubuntu)
https://travis-ci.org/nunit/nunit/jobs/322775656#L728 (OS X)

--- a/src/Cake.Core/Scripting/ScriptConventions.cs
+++ b/src/Cake.Core/Scripting/ScriptConventions.cs
@@ -84,6 +84,7 @@ namespace Cake.Core.Scripting
 
             // This is just to please Roslyn when running under Mono. See issue https://github.com/dotnet/roslyn/issues/19364
             result.Add(_loader.Load(new AssemblyName("System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"))); // System.Runtime
+            result.Add(_loader.Load(new AssemblyName("System.Collections, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"))); // System.Collections
 #endif
 
             // Return the assemblies.

--- a/tests/integration/Cake.Core/Scripting/SystemCollections.cake
+++ b/tests/integration/Cake.Core/Scripting/SystemCollections.cake
@@ -1,0 +1,10 @@
+Task("Cake.Core.Scripting.SystemCollections.ToDictionary")
+    .Does(() =>
+{
+    var dictionary = Array.Empty<int>().ToDictionary(key => key, value=>value);
+});
+
+//////////////////////////////////////////////////////////////////////////////
+
+Task("Cake.Core.Scripting.SystemCollections")
+    .IsDependentOn("Cake.Core.Scripting.SystemCollections.ToDictionary");

--- a/tests/integration/build.cake
+++ b/tests/integration/build.cake
@@ -22,6 +22,7 @@
 #load "./Cake.Common/Tools/NuGet/NuGetAliases.cake"
 #load "./Cake.Core/Scripting/DefineDirective.cake"
 #load "./Cake.Core/Scripting/LoadDirective.cake"
+#load "./Cake.Core/Scripting/SystemCollections.cake"
 #load "./Cake.Core/Scripting/UsingDirective.cake"
 #load "./Cake.Core/Tooling/ToolLocator.cake"
 #load "./Cake.Core/CakeAliases.cake"
@@ -48,6 +49,7 @@ Setup(ctx =>
 Task("Cake.Core")
     .IsDependentOn("Cake.Core.Scripting.DefineDirective")
     .IsDependentOn("Cake.Core.Scripting.LoadDirective")
+    .IsDependentOn("Cake.Core.Scripting.SystemCollections")
     .IsDependentOn("Cake.Core.Scripting.UsingDirective")
     .IsDependentOn("Cake.Core.Tooling.ToolLocator")
     .IsDependentOn("Cake.Core.CakeAliases");

